Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryAuditTransforms.java

```diff
@@ -129,19 +130,42 @@ private II getPatientIdFromRequest(PRPAIN201305UV02 request) {
             && request.getControlActProcess().getQueryByParameter().getValue() != null
             && request.getControlActProcess().getQueryByParameter().getValue().getParameterList() != null
             && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId() != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0) != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0).getValue() != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0).getValue().get(0) != null) {
-
-            oII = request.getControlActProcess().getQueryByParameter().getValue().getParameterList().
-                getLivingSubjectId().get(0).getValue().get(0);
+            .getLivingSubjectId() != null) {
+
+            List<PRPAMT201306UV02LivingSubjectId> ids = request.getControlActProcess().getQueryByParameter().getValue()
+                .getParameterList().getLivingSubjectId();
+
+            if (ids.size() == 1) {
+                livingSubjectId = getLivingSubjectId(ids.get(0));
+            } else if (ids.size() > 1) {
+                // Get assignedDevice root
+                String root = request.getControlActProcess().getAuthorOrPerformer().get(0).getAssignedDevice()
+                    .getValue().getId().get(0).getRoot();
+
+                // Compare assignedDevice root to each livingSubjectId root
```

Comment:
Also, might have wanted to pull out the matching of the IDs to the root into another method. Would make the code clearer and eliminate some of the cyclomatic complexity (I think it gets to 4 levels, not too bad)